### PR TITLE
python3Packages.zephyr-python-api: init at 0.0.3

### PR DESCRIPTION
**Pull request by Steinhagen**
###### Description of changes

Add Python package zephyr-python-api.

Homepage: https://github.com/nassauwinter/zephyr-python-api

This is a set of wrappers for Zephyr Scale (TM4J) REST API. This means you can interact with Zephyr Scale without GUI, access it with python code and create automation scripts for your every day interactions.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (natsukium) [inline maintainers/maintainer-list.nix:1]: Would you please split this change into a separate commit?

File: pkgs/development/python-modules/zephyr-python-api/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib
+, requests
+, buildPythonPackage
+, fetchPypi
+, setuptools
```
Comment (natsukium) [inline pkgs/development/python-modules/zephyr-python-api/default.nix:5]:
```suggestion
, buildPythonPackage
, fetchPypi
, requests
, setuptools
```
Usually, we sort inputs alphabetically.

File: pkgs/development/python-modules/zephyr-python-api/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    requests
+  ];
+
+  nativeBuildInputs = [ setuptools ];
+
+  pythonImportsCheck = [ "zephyr" ];
+
+  meta = {
+    homepage = "https://github.com/nassauwinter/zephyr-python-api";
+    description = "This is a set of wrappers for Zephyr Scale (TM4J) REST API.";
```
Comment (natsukium) [inline pkgs/development/python-modules/zephyr-python-api/default.nix:28]:
```suggestion
    description = "A set of wrappers for Zephyr Scale (TM4J) REST API";
```
We should remove a trailing period and redundant expression.
https://nixos.org/manual/nixpkgs/stable/#var-meta-description

**Review by natsukium — CHANGES_REQUESTED** [review]
Would you please read [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md) and rewrite the commit message and a PR title?
Thanks!

File: pkgs/development/python-modules/zephyr-python-api/default.nix (hunk 1 above)
Comment (Steinhagen) [reply pkgs/development/python-modules/zephyr-python-api/default.nix]: Done.

File: pkgs/development/python-modules/zephyr-python-api/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "zephyr-python-api";
+  version = "0.0.3";
+  format = "pyproject";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "33d29fd11b6849e0c5000800ba4b3e124f9683938cf2a99ddde0e86808006b70";
```
[the remark this answers is not in the document]
Comment (Steinhagen) [reply pkgs/development/python-modules/zephyr-python-api/default.nix]: Done.

File: pkgs/development/python-modules/zephyr-python-api/default.nix (hunk 2 above)
Comment (Steinhagen) [reply pkgs/development/python-modules/zephyr-python-api/default.nix]: Fixed.

File: maintainers/maintainer-list.nix
Comment (natsukium) [reply maintainers/maintainer-list.nix:1]: Could you please check this?

File: pkgs/development/python-modules/zephyr-python-api/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  propagatedBuildInputs = [
+    requests
+  ];
+
+  nativeBuildInputs = [ setuptools ];
+
```
[the remark this answers is not in the document]
Comment (Steinhagen) [reply pkgs/development/python-modules/zephyr-python-api/default.nix:26]: There is no need to add pytestCheckHook. The archive doesn't pack any test.

File: maintainers/maintainer-list.nix
Comment (Steinhagen) [reply maintainers/maintainer-list.nix:1]: This was added as a separate commit. The current PR doesn't add me as a maintainer anymore.
This was done in PR: https://github.com/NixOS/nixpkgs/pull/248307

**Comment by Steinhagen** [comment]
I believe all reported issues were resolved.

File: pkgs/development/python-modules/zephyr-python-api/default.nix (hunk 2 above)
Comment (natsukium) [reply pkgs/development/python-modules/zephyr-python-api/default.nix]: The period still remains.

Comment (Steinhagen) [reply pkgs/development/python-modules/zephyr-python-api/default.nix]: > The period still remains.

Just fixed this. Thanks for looking into this!

**Comment by natsukium** [comment]
@ofborg build python310Packages.zephyr-python-api

**Review by natsukium — APPROVED** [review]
LGTM, thanks!
